lib/opensusebasetest.pm

```diff
@@ -360,8 +360,17 @@ sub wait_boot {
     unlock_if_encrypted if !get_var('S390_ZKVM');
 
     if ($textmode || check_var('DESKTOP', 'textmode')) {
-        assert_screen [qw(linux-login emergency-shell emergency-mode)], $ready_time;
+        my $matched_needle = check_screen [qw(linux-login emergency-shell emergency-mode)], $ready_time;
         handle_emergency if (match_has_tag('emergency-shell') or match_has_tag('emergency-mode'));
+
+        #Soft-fail for user_defined_snapshot in extra_tests_on_gnome and extra_tests_on_gnome_on_ppc
+        if (!$matched_needle && get_var('TEST') =~ "extra_tests_on_gnome") {
```

Comment:
because you just replaced assert_screen with check_screen this 'if'  needs 'else' branch to cover behavior what to do when !$matched_needle but it is not "extra_tests_on_gnome"